c7n/resources/elbv2.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2016 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+V2 Elastic Load Balancers
+"""
+import logging
+
+from c7n.actions import ActionRegistry, BaseAction
+from c7n.filters import Filter, FilterRegistry, DefaultVpcBase
+from c7n import tags
+from c7n.manager import resources
+from c7n.query import QueryResourceManager
+from c7n.utils import local_session, chunks, type_schema, get_retry
+
+log = logging.getLogger('custodian.elbv2')
+
+filters = FilterRegistry('elbv2.filters')
+actions = ActionRegistry('elbv2.actions')
+
+filters.register('tag-count', tags.TagCountFilter)
+filters.register('marked-for-op', tags.TagActionFilter)
+
+
+@resources.register('elbv2')
+class ELBV2(QueryResourceManager):
+    """Resource manager for v2 ELBs (AKA ALBs).
+    """
+
+    class Meta(object):
+
+        service = 'elbv2'
+        type = 'elbv2'
+        enum_spec = ('describe_load_balancers', 'LoadBalancers', None)
+        name = 'LoadBalancerName'
+        id = 'LoadBalancerArn'
+        filter_name = None
+        filter_type = None
+        dimension = None
+        date = 'CreatedTime'
+
+    resource_type = Meta
+    filter_registry = filters
+    action_registry = actions
+    retry = staticmethod(get_retry(('Throttling',)))
+
+    def augment(self, albs):
+        _describe_elbv2_tags(
+            albs, self.session_factory, self.executor_factory, self.retry)
+        _describe_elbv2_listeners(
```

Comment:
i'd prefer we hide this behind separate filters, to avoid the additional queries unless someone is using the data. ditto for groups. especially since its a per resource api call, and trying to avoid rate limit problems. thoughts?